pkg/apis/performanceprofile/v2/performanceprofile_validation.go

```diff
@@ -105,9 +129,22 @@ func (r *PerformanceProfile) validateNodeSelectorDuplication(ppList *Performance
 func (r *PerformanceProfile) ValidateBasicFields() field.ErrorList {
 	var allErrs field.ErrorList
 
+	nodes, err := r.getNodesList()
+	if err != nil {
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("spec.nodeSelector"),
+				r.Spec.NodeSelector,
+				err.Error(),
+			),
+		)
+	}
+
 	allErrs = append(allErrs, r.validateCPUs()...)
 	allErrs = append(allErrs, r.validateSelectors()...)
-	allErrs = append(allErrs, r.validateHugePages()...)
+	allErrs = append(allErrs, r.validateAllNodesAreSameCpuArchitecture(nodes)...)
+	allErrs = append(allErrs, r.validateAllNodesAreSameCpuCapacity(nodes)...)
```

Comment:
I'm concerned that this might be too restrictive - added another comment to discuss.

Reply:
All nodes in each MCP must be the same architecture and the same CPU topology. That is documented and even somewhat enforced by NTO (it would generate two separate kernel args sets for the different nodes and NTO would stop applying that due to conflict).

The validation just catches that with a nicer message.

@bartwensley Do you see a flow where this assumption would not be valid?

Reply:
OK - thanks Martin - makes sense.